test/functions/test_openai_function.py

```diff
@@ -50,7 +51,73 @@ def test_all_parameters(
             default_enum_para (RoleType): default_enum_para desc
         """
 
-    expect_res = {
+    # pydantic v1 follows JSON Schema Draft-07 and v2 now pin to 2020-12
+    # ref: https://github.com/pydantic/pydantic/issues/4666
+    expect_res_v1 = {
```

Comment:
What is the difference between `expect_res_v2` and `expect_res_v2 `? Does the difference in the json schema effort the function calling?

Reply:
> What is the difference between `expect_res_v2` and `expect_res_v2 `? Does the difference in the json schema effort the function calling?

I think you mean `v1` vs `v2`? Very small changes, like `definitions` in v1 becomes `$defs` in v2, it provides a function abstraction.